core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -275,15 +275,14 @@ private static List<Transaction> genesisTransactions() {
 
     @Override
     public int messageSize() {
-        int size = HEADER_SIZE;
         List<Transaction> transactions = getTransactions();
-        if (transactions != null) {
-            size += VarInt.sizeOf(transactions.size()) +
-                    transactions.stream()
-                            .mapToInt(Transaction::messageSize)
-                            .sum();
-        }
-        return size;
+        return (transactions == null)
+                ? HEADER_SIZE
```

Comment:
Add HEADER_SIZE first, and not inside the terniary? That would be more to the point.